File: 821-shortest-distance-to-a-character/821-shortest-distance-to-a-character.py
class Solution:
    def shortestToChar(self, s: str, c: str) -> List[int]:
        arr = []
        for i in range(len(s)):
            if s[i] == c:
                arr.append(i)

        newarr = []
        for i in range(len(s)):
            minnum = len(s)
            for j in range(len(arr)):
                if abs(i-arr[j]) < minnum:
                    minnum = abs(i-arr[j])
            newarr.append(minnum)
        return newarr    

File: 821-shortest-distance-to-a-character/821-shortest-distance-to-a-character.java
class Solution {
    public int[] shortestToChar(String s, char c) {
        List<Integer> arr = new ArrayList<>();
        for (int i = 0; i < s.length();i++){
            if (s.charAt(i) == c){
                arr.add(i);
            }
        }
        int[] newArr = new int[s.length()];
        for (int i = 0; i < s.length();i++){
            int minimum = s.length();
            for (int j = 0; j < arr.size();j++){
                if (Math.abs(i-arr.get(j)) < minimum){
                    minimum = Math.abs(i-arr.get(j));
                }
            }
            newArr[i] = minimum;
        }
        return newArr;   
    }
}